source/oauth/settings.d

```diff
@@ -406,20 +406,20 @@ class OAuthSettings
         Params:
             scopes = An array of identifiers specifying the scope of
                 access to be requested. (optional)
+            params = Additional parameters of request. (optional)
 
         Returns: The new session.
 
         Throws: OAuthException if authentication fails.
       +/
     final
-    OAuthSession clientSession(string[] scopes = null) immutable
+    OAuthSession clientSession(string[] scopes = null, string[string] params = null) immutable
```

Comment:
The ability to send extra parameters seems indeed necessary for vk.com. Though some precautions would be necessary to prevent misuse by setting parameters that should be controlled by the oauth library only. An attempt to do so should result in an exception.

Also, this should probably be an extra overload on `clientSession`, both to not bother the general case with the additional checking, and to be able to switch parameter order to be more consistent with `userAuthUri`. To prevent code duplication, that would mean turning this into a private method with two `public final` wrappers.

Reply:
Found how to bypass this without additional `params` argument. I removed this code.